### PR TITLE
feat(deploy): add support for updating values in YAML arrays

### PR DESCRIPTION
**Pull request by christiansiegel**
Support YAML paths containing array indexes, e.g. `a.b.[1].c` for 
```yaml
a:
  b:
  - foo
  - c: bar
`

### CODE REVIEW

File: gitopscli/yaml/yaml_util.py [hunk 1]
```diff
@@ -20,23 +23,32 @@ def get_string(self):
...
-            if not create_new:
-                raise KeyError(f"Key '{key}' not found in YAML!")
-            obj[k] = dict()
-        obj = obj[k]
-    if keys[-1] in obj and obj[keys[-1]] == value:
-        return False  # nothing to update
-    if not create_new and keys[-1] not in obj:
-        raise KeyError(f"Key '{key}' not found in YAML!")
-    obj[keys[-1]] = value
+    keys, item = key.split("."), content
```
Comment (niiku) [inline gitopscli/yaml/yaml_util.py:31]: Noob Question: Why assign content to item? Is there some magic I'm missing here?
Edit: I see it now - maybe provide a better name for `item` as it's a bit irritating (something like `current_item`) so it's clear that the reference is changing in the loop.

File: gitopscli/yaml/yaml_util.py [hunk 2]
```diff
@@ -20,23 +23,32 @@ def get_string(self):
...
+    leaf_idx = len(keys) - 1
+    current_key_segments = []
+    current_key = ""
```
Comment (niiku) [inline gitopscli/yaml/yaml_util.py:34]: Is there a reason why current_key is already defined? Isn't in the loop itself sufficient?

File: gitopscli/yaml/yaml_util.py [hunk 3]
```diff
@@ -20,23 +23,32 @@ def get_string(self):
...
+                raise KeyError(f"Key '{current_key}' not found in YAML!")
+        if i == leaf_idx:
+            if item[k] == value:
+                return False  # nothing to update
+            item[k] = value
+            break
+        item = item[k]
 
     with open(file_path, "w+") as stream:
         yaml.dump(content, stream)
```
Comment (niiku) [inline gitopscli/yaml/yaml_util.py:54]: Maybe replace the `dump` and `return True` with the `break` on line 50? So the flow of the program is more clear

File: gitopscli/yaml/yaml_util.py (hunk 1 above)
Comment (christiansiegel) [reply gitopscli/yaml/yaml_util.py]: actually `parent_item` in the context of the loop. fixed.

File: gitopscli/yaml/yaml_util.py (hunk 2 above)
Comment (christiansiegel) [reply gitopscli/yaml/yaml_util.py]: you're right. removed

File: gitopscli/yaml/yaml_util.py (hunk 3 above)
Comment (christiansiegel) [reply gitopscli/yaml/yaml_util.py]: ack :+1:

**Comment by niiku** [comment]
lgtm - I guess you can merge

**Pull request merged**